pkg/rosa/nodepools.go

```diff
@@ -0,0 +1,67 @@
+package rosa
+
+import cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+func (c *rosaClient) CreateNodePool(clusterID string, nodePool *cmv1.NodePool) (*cmv1.NodePool, error) {
```

Comment:
Why do we need these wrappers?

Reply:
Well, the ocm sdk is not really intuitive to use directly.
Also it helps keep the controller reconciliation code cleaner. We should actually move the building of the Cluster and NodePool ocm objects out of the controller as well.

Reply:
Perhaps we send a PR to the SDK repo? I don't think the upstream cluster-api-provider-aws repo is a good place to wrap the SDK to make it more intuitive.

Reply:
This is not just a wrapper, it provides reusable functions, like `GetCluster()` and `handleError()`. It serves to decuple the controller logic from the implementation details of the underlying sdk (ocm). As I said we should be moving all ocm specific logic here, like constructing the ocm cluster spec.
Obviously this will contain more helpful functionalities as we add more features.

Reply:
Generic error handling in a wrapper seems like a useful and novel addition - no argument there.

As for the rest of it, I would question the necessity. This is a leaky abstraction that requires an ever-increasing parameter list to handle the options in the SDK that the users need, or copy-paste. See your `GetNodePools` and `GetNodePool` methods. Why do we need both? Is it that onerous to pass an integer to the SDK?

As a consideration - why don't we wrap a `*kubernetes.Clientset` with some method like

```go
type wrapper struct {
    delegate *kubernetes.Clientset
}

func (w *wrapper) GetPod(namespace, name string) (*corev1.Pod, error) {
    return w.delegate.Core().V1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
}
```

From my perspective, if `*rosaClient` is justified, the `*wrapper` above would be, too.

Let's get the PR in but reconsider if we truly need this in the final versions.